WordPress/WordPressTest/CoreDataMigrationTests.m

```diff
@@ -967,6 +967,100 @@ - (void)testMigrate74to75
+
+    NSManagedObject *account1 = [NSEntityDescription insertNewObjectForEntityForName:@"Account" inManagedObjectContext:context];
+    [account1 setValue:@"dotcomuser1" forKey:@"username"];
+
+    //WPCom Blog
+    NSManagedObject *blog1 = [NSEntityDescription insertNewObjectForEntityForName:@"Blog" inManagedObjectContext:context];
+    [blog1 setValue:@(1001) forKey:@"blogID"];
+    [blog1 setValue:@"https://test1.wordpress.com" forKey:@"url"];
+    [blog1 setValue:@"https://test1.wordpress.com/xmlrpc.php" forKey:@"xmlrpc"];
+    [blog1 setValue:account1 forKey:@"account"];
+
+    //SelfHosted Blog
+    NSManagedObject *blog2 = [NSEntityDescription insertNewObjectForEntityForName:@"Blog" inManagedObjectContext:context];
+    [blog2 setValue:@(1002) forKey:@"blogID"];
+    [blog2 setValue:@"https://test1.wordpress.com" forKey:@"url"];
+    [blog2 setValue:@"https://test1.wordpress.com/xmlrpc.php" forKey:@"xmlrpc"];
+
+    [context save:&error];
+    XCTAssertNil(error, @"Error while saving context");
+
+    // Cleanup
+    XCTAssertNotNil(ps);
+    psc = nil;
+
+    // Migrate to Model 88
+    NSManagedObjectModel *model88 = [[NSManagedObjectModel alloc] initWithContentsOfURL:model88Url];
+    BOOL migrateResult = [ALIterativeMigrator iterativeMigrateURL:storeUrl
+                                                           ofType:NSSQLiteStoreType
+                                                          toModel:model88
+                                                orderedModelNames:@[@"WordPress 87", @"WordPress 88"]
+                                                            error:&error];
+    if (!migrateResult) {
+        NSLog(@"Error while migrating: %@", error);
+    }
+    XCTAssertTrue(migrateResult);
+
+    // Load a Model 88 Stack
+    psc = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:model88];
+    ps = [psc addPersistentStoreWithType:NSSQLiteStoreType
+                           configuration:nil
+                                     URL:storeUrl
+                                 options:options
+                                   error:&error];
+
+    context = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSMainQueueConcurrencyType];
+    context.persistentStoreCoordinator = psc;
+
+    NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:@"Blog"];
+    request.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"blogID" ascending:YES]];
+
+    NSArray *results = [context executeFetchRequest:request error:nil];
+    XCTAssert(results.count == 2, @"Error Fetching Blogs");
+
+    // WPCom/Jetpack sites should migrate to Gutenberg
+
+    NSManagedObject *wpcomBlog = [results firstObject];
+    XCTAssertEqualObjects([wpcomBlog valueForKey:@"mobileEditor"], @"gutenberg");
+
+    // SelfHosted (non-jetpack) should migrate to Aztec
+
+    NSManagedObject *selfHostedBlog = [results lastObject];
+    XCTAssertEqualObjects([selfHostedBlog valueForKey:@"mobileEditor"], @"aztec");
```

Comment:
After running the app on the simulator, this test fails for me because the migration is setting the value according to `kUserDefaultsGutenbergEditorEnabled`

Reply:
Right. We need user defaults for the migration.

The first thing that comes into mind is to use user default from the Environment introduced in https://github.com/wordpress-mobile/WordPress-iOS/pull/12196

Do you have other options?

Reply:
That's probably the sensible route to take. I thought of maybe trying to inject the `database` into the `userInfo` of `NSMigrationManager`, but that would require too many changes to `ALIterativeMigrator` and I don't like the risk of breaking things there.

Maybe `OCMock` can replace the standard user defaults, since the tests are Obj-C?